connection.py

```diff
@@ -0,0 +1,12 @@
+class Connection():
+
+    def __init__(self, pos1, pos2):
```

Comment:
I suggest calling the parameters the same things as the member variables. it might a little confusing that in one case they are "pos" and in another "endpoint"